### PR TITLE
wayland: Stable protocols can have a version

### PR DESCRIPTION
**Pull request by xclaesse**
Fixes: #12968

### CODE REVIEW

File: mesonbuild/modules/wayland.py [hunk 1]
```diff
@@ -122,20 +122,20 @@ def find_protocol(self, state: ModuleState, args: T.Tuple[str], kwargs: FindProt
...
         if self.pkgdatadir is None:
             self.pkgdatadir = self.protocols_dep.get_variable(pkgconfig='pkgdatadir', internal='pkgdatadir')
 
-        if xml_state == 'stable':
-            xml_name = f'{base_name}.xml'
-        elif xml_state == 'staging':
-            xml_name = f'{base_name}-v{version}.xml'
-        else:
-            xml_name = f'{base_name}-unstable-v{version}.xml'
+        xml_name = basename
```
Comment (bilelmoussaoui) [inline mesonbuild/modules/wayland.py:133]:
```suggestion
        xml_name = base_name
```

**Review by bilelmoussaoui — COMMENTED** [review]
otherwise the build seems to work now, thanks a lot for the quick fix

File: mesonbuild/modules/wayland.py (hunk 1 above)
Comment (xclaesse) [reply mesonbuild/modules/wayland.py]: done

**Review by tristan957 — APPROVED** [review]

**Comment by tristan957** [comment]
One thought. It would be good to add a test! Shouldn't be too hard :).

**Comment by xclaesse** [comment]
> One thought. It would be good to add a test! Shouldn't be too hard :).

The test loads protocols from system and non has a version currently.

**Pull request merged**